src/process.h

```diff
@@ -290,6 +313,29 @@ class Process : public ProcessListFromProcessGroup::Element,
 
   Profiler* profiler_ = null;
 
+  Object* marked_allocation_failed_;
```

Comment:
It would be nice if the interpreter could create these from a table in the program -- and then we can let the primitives themselves return the (marked) indexes into the table marked as the failure.